fury/ui/elements.py

```diff
@@ -3272,3 +3272,696 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+
+    def _set_position(self, _coords):
+        """ Position the lower-left corner of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        self.base_node.position = _coords
+
+    def _get_size(self):
+        return self.base_node.size
+
+    def generate_tree(self, structure):
+        """Generate the structure of the Tree2D
+
+        Parameters
+        ----------
+        structure: List(Dict, Dict, .....)
+            [{'node': ['child', 'child']}, .....]
+        """
+        for obj in structure:
+            parent_label = list(obj.keys())[0]
+            child_labels = list(obj.values())[0]
+
+            if parent_label in self.nodes_dict.keys():
+                parent_node = self.nodes_dict[parent_label]
+            else:
+                parent_node = TreeNode2D(label=parent_label)
+                self._nodes.append(parent_node)
+                self._nodes_dict[parent_label] = parent_node
+
+            child_nodes = [TreeNode2D(label=child_label) for child_label in
+                           child_labels]
+
+            for child_node, child_label in zip(child_nodes, child_labels):
+                self._nodes_dict[child_label] = child_node
+                parent_node.add_node(child_node)
+
+    def resize(self, size):
+        """ Resizes the Tree Node.
+
+        Parameters
+        ----------
+        size : (int, int)
+            New width and height in pixels.
+        """
+        self.base_node.resize(size)
+
+    def add_content(self, node, content, coords=(0., 0.)):
+        """Add content to a sepcific node
+
+        Parameters
+        ----------
+        node: str
+            Label of the node
+        content: :class: `UI` or `TreeNode2D`
+            The content that is to be added in the node
+        coords : (float, float) or (int, int), optional
+            If float, normalized coordinates are assumed and they must be
+            between [0,1].
+            If int, pixels coordinates are assumed and it must fit within the
+            content panel's size.
+        """
+        _node = self.select_node(node)
+        _node.add_node(content, coords)
+
+        _node.set_visibility(False)
+
+    def select_node(self, node):
+        """Get the instance of a specific node
+
+        Parameters
+        ----------
+        node: str
+            Label of the node
+        """
+        _node = self.nodes_dict[node]
+        return _node
+
+    @property
+    def nodes(self):
+        """Get all the nodes present in the Tree2D
+
+        Returns
+        -------
+        list of nodes
+        """
+        return self._nodes
+
+    @property
+    def nodes_dict(self):
+        """Get all the nodes present in the Tree2D in dict format
+
+        Returns
+        -------
+        dict with label, node as key, value
+        """
+        return self._nodes_dict
+
+
+class TreeNode2D(UI):
```

Comment:
you should add `TreeNode2D` in the `__all__` at the top of the file

Reply:
Done